p2p/pex/pex_reactor.go

```diff
@@ -752,10 +818,33 @@ func (r *Reactor) attemptDisconnects() {
 
 func markAddrInBookBasedOnErr(addr *p2p.NetAddress, book AddrBook, err error) {
 	// TODO: detect more "bad peer" scenarios
+	// TODO, blacklist peer if fmt.Errorf("peer is on a different network. Got %v, expected %v", other.Network, info.Network)
 	switch err.(type) {
 	case p2p.ErrSwitchAuthenticationFailure:
 		book.MarkBad(addr, defaultBanTime)
 	default:
 		book.MarkAttempt(addr)
 	}
 }
+
+func (r *Reactor) TooEarlyToDial(addr *p2p.NetAddress, attempts int, lastDialedTime time.Time) error {
```

Comment:
I would merge these two methods, right. All of them are about number of dial attempts.

Reply:
Combined here https://github.com/cometbft/cometbft/pull/3371/commits/7c12ee7fbbb1dfc38bf2f285060410693d3cfd39